src/omv/imlib/filter.c

```diff
@@ -1325,53 +1324,161 @@ void imlib_morph(image_t *img,
         case PIXFORMAT_GRAYSCALE: {
             buf.data = fb_alloc(IMAGE_GRAYSCALE_LINE_LEN_BYTES(img) * brows, FB_ALLOC_NO_HINT);
 
-            for (int y = 0, yy = img->h; y < yy; y++) {
+            #if defined(ARM_MATH_DSP)
+            int32_t krn_4, krn_2_0, krn_5_3, krn_8_6, krn_7_1, offset_int, invert_ge, invert_lt;
+            if (ksize == 1) {
+                krn_4 = krn[4];
+                krn_2_0 = __PKHBT(krn[0], krn[2], 16);
+                krn_5_3 = __PKHBT(krn[3], krn[5], 16);
+                krn_8_6 = __PKHBT(krn[6], krn[8], 16);
+                krn_7_1 = __PKHBT(krn[1], krn[7], 16);
+                offset_int = __PKHBT(offset, offset, 16);
+                invert_ge = invert ? 0x00FF00FF : 0xFF00FF00;
+                invert_lt = invert ? 0xFF00FF00 : 0x00FF00FF;
```

Comment:
Krn can be moved into an array without any speed impact. But, I didn't put it into an array so it was easier to see which value is which when writing the code since the values are moving around not necessarily linearly.

I'm not sure what your comment about the static inline function is. All these values are directly passed to compiler intrinsics in the tight loop code.

Reply:
Since this block is repeated twice, and the other one 3 times, I just suggesting you replace it with an inline function or a macro, for readability and to avoid copy&paste errors, but I guess it's okay since it's only used in this single function, but please use arrays instead of `int32_t krn_4, krn_2_0, krn_5_3, krn_8_6, krn_7_1` this should be an easy search&replace.

Reply:
Same as above. Per our conversation in meeting.